src/sentry/search/events/builder.py

```diff
@@ -1835,7 +1836,12 @@ def resolve_snql_function(
     def resolve_metric_index(self, value: str) -> Optional[int]:
         """Layer on top of the metric indexer so we'll only hit it at most once per value"""
         if value not in self._indexer_cache:
-            result = indexer.resolve(self.organization_id, value)
+            use_case_id = UseCaseKey.PERFORMANCE
```

Comment:
Your comment is a bit vague, what do you mean?

Reply:
I'll remove it of course, so what i'm trying to understand is -> we still can invoke this method `resolve_metric_index ` for both cases: `UseCaseKey.PERFORMANCE` and `UseCaseKey.RELEASE_HEALTH`.

So i'm trying to find a way to pass it somehow - or tweak tests.

Reply:
I don't think this module should invoke release health, why are you saying that? Does the alert code use this?

Reply:
Correct, it was my first impression. But there few tests that are doing it from code.

Reply:
removed comment from code.

Reply:
if you checkout my branch locally and run: `pytest tests/sentry/snuba/test_tasks.py -k test_query_and_environment_users_metrics` you can see:
```
―――――――――――――――――――――――――――――――――――――――――――――――― BuildSnqlQueryTest.test_query_and_environment_users_metrics ――――――――――――――――――――――――――――――――――――――――――――――――
tests/sentry/snuba/test_tasks.py:1006: in test_query_and_environment_users_metrics
    self.run_test(
tests/sentry/snuba/test_tasks.py:456: in run_test
    query_builder = build_query_builder(
src/sentry/snuba/tasks.py:187: in build_query_builder
    return entity_subscription.build_query_builder(query, project_ids, environment, params)
src/sentry/snuba/entity_subscription.py:345: in build_query_builder
    qb = AlertMetricsQueryBuilder(
src/sentry/search/events/builder.py:2160: in __init__
    super().__init__(*args, **kwargs)
src/sentry/search/events/builder.py:1608: in __init__
    super().__init__(
src/sentry/search/events/builder.py:164: in __init__
    self.resolve_query(
src/sentry/search/events/builder.py:1641: in resolve_query
    self.where, self.having = self.resolve_conditions(
src/sentry/search/events/builder.py:312: in resolve_conditions
    where = self.resolve_where(parsed_terms)
src/sentry/search/events/builder.py:270: in resolve_where
    condition = self.format_search_filter(term)
src/sentry/search/events/builder.py:949: in format_search_filter
    converted_filter = self.convert_search_filter_to_condition(
src/sentry/search/events/builder.py:1007: in convert_search_filter_to_condition
    return converter(search_filter)
src/sentry/search/events/builder.py:1869: in _default_filter_converter
    value = self._resolve_tag_value(value)
src/sentry/search/events/builder.py:1850: in _resolve_tag_value
    raise InvalidSearchQuery("Tag value was not found")
E   sentry.exceptions.InvalidSearchQuery: Tag value was not found
```
then if you apply:
```diff
diff --git src/sentry/search/events/builder.py src/sentry/search/events/builder.py
index 314866d4fa..7b6f01f3e5 100644
--- src/sentry/search/events/builder.py
+++ src/sentry/search/events/builder.py
@@ -1836,7 +1836,7 @@ class MetricsQueryBuilder(QueryBuilder):
     def resolve_metric_index(self, value: str) -> Optional[int]:
         """Layer on top of the metric indexer so we'll only hit it at most once per value"""
         if value not in self._indexer_cache:
-            use_case_id = UseCaseKey.PERFORMANCE
+            use_case_id = UseCaseKey.RELEASE_HEALTH
             result = indexer.resolve(self.organization_id, value, use_case_id=use_case_id)
             self._indexer_cache[value] = result
```
and rerun test it works.

Reply:
Ah okay.. I understand now, in that case you probably want to check if this query builder is building a query for crash rate alerts which are session alerts and then set release health case. Maybe check with @wmak to help you navigate that part of the code?

Reply:
You should be able to check `self.dataset is Dataset.Metrics` for release health and `self.dataset is Dataset.PerformanceMetrics` for MEP

If you're feeling keen we could add a class property that does this as well


```
@property
def is_performance_metrics(self):
    return self.dataset is Dataset.PerformanceMetrics
```

Reply:
thanks! yea i were thinking about similar approach.

Reply:
@wmak one small quiestion:
as you can see in this data class:
```
@unique
class Dataset(Enum):
    Events = "events"
    Transactions = "transactions"
    Discover = "discover"
    Outcomes = "outcomes"
    OutcomesRaw = "outcomes_raw"
    Sessions = "sessions"
    # Actually Release Health
    Metrics = "metrics"
    PerformanceMetrics = "generic_metrics"
```
i'm not fully sure how do we want to distinguish all data class fields into 2 sub-groups for `indexer`:
- release_health and performance
---
Previously we discussed with @ahmedetefy that we have 2 groups: 
`transactions` are belongs to `UseCaseKey.PERFORMANCE` value for indexer and
`sessions` are belongs to `UseCaseKey.RELEASE_HEALTH`.

Reply:
You don't have to handle all the Datasets, the `MetricQueryBuilder`s only accept two:
https://github.com/getsentry/sentry/blob/7f6c5f9e89c0ceb49e2d89959ca4781cdd0942e1/src/sentry/search/events/builder.py#L1607